pennylane/io/qasm_interpreter.py

```diff
@@ -829,8 +838,33 @@ def visit_quantum_reset(self, node: QASMNode, context: dict):
             node (QASMNode): the quantum reset node.
             context (dict): the current context.
         """
-        wire = self.visit(node.qubits, context)
-        measure(context.wire_map.get(wire, wire), reset=True)
+        wires = self.visit(node.qubits, context)
+        if isinstance(wires, list):
+            for wire in wires:
+                measure(context.wire_map.get(wire, wire), reset=True)
+        else:
+            measure(context.wire_map.get(wires, wires), reset=True)
+
+    @staticmethod
+    def _bind_quantum_parameter(param, evald_arg, inner_context, context):
+        """
+        Binds a quantum parameter in a subroutine or custom gate context.
+
+        Args:
+            param (str): the name of the quantum parameter to bind.
+            evald_arg (str): the name of the qubit.
+            inner_context (Context): the custom gate or subroutine context.
+            context (Context): the outer context.
+        """
+        reg = False
+        if evald_arg in context.registers:
+            reg = True
+            del inner_context.wires[inner_context.wires.index(param)]
```

Comment:
Just to stick to the API provided by the data structures we are using (correct me if I'm wrong but `inner_context.wires` is a list, right?):
```suggestion
            inner_context.wires.pop(inner_context.wires.index(param))
```

Reply:
Hi Mudit, thanks for your feedback. I would argue `del` is a little bit more appropriate since we are not interested in returning the removed item here.